components/director/internal/domain/application/resolver.go

```diff
@@ -1344,6 +1340,26 @@ func (r *Resolver) getApplication(ctx context.Context, get func(context.Context)
 	return r.appConverter.ToGraphQL(app), nil
 }
 
+func (r *Resolver) getOperation(ctx context.Context, obj *graphql.Application) (*model.Operation, error) {
+	op, err := r.opSvc.GetByDataAndType(ctx, data.NewOrdOperationData(obj.ID, ""), model.OperationTypeOrdAggregation)
+	if err == nil {
+		return op, nil
+	}
+
+	if apperrors.IsNotFoundError(err) {
+		if obj.ApplicationTemplateID != nil {
+			op, err = r.opSvc.GetByDataAndType(ctx, data.NewOrdOperationData(obj.ID, *obj.ApplicationTemplateID), model.OperationTypeOrdAggregation)
+			if err == nil || apperrors.IsNotFoundError(err) {
+				return op, nil
+			}
+		} else {
+			return op, nil
+		}
+	}
+
+	return nil, errors.Wrap(err, "while getting operation")
+}
```

Comment:
Here's a simpler function but I may be missing a case:

```
func (r *Resolver) getOperation(ctx context.Context, obj *graphql.Application) (*model.Operation, error) {
	appTemplateID := ""
	if obj.ApplicationTemplateID != nil {
		appTemplateID = *obj.ApplicationTemplateID
	}

	op, err := r.opSvc.GetByDataAndType(ctx, data.NewOrdOperationData(obj.ID, appTemplateID), model.OperationTypeOrdAggregation)
	if err != nil && !apperrors.IsNotFoundError(err) {
		return nil, errors.Wrap(err, "while getting operation")
	}

	return op, nil
}
```

Reply:
This was exactly the code before the change. Here if the application has an app template we will search for operation with appID and appTemplate id - this means that the processing will be for the application in the context of app template's ord webhook, but there is a case where an application that has an app template, but also has an ord webhook and the processing must be in the context of the application's ord webhook.